sqlite/vdbe.c

```diff
@@ -2495,7 +2495,7 @@ case OP_Ge: {             /* same as TK_GE, jump, in1, in3 */
     pOut = &aMem[pOp->p2];
     iCompare = res;
     res2 = res2!=0;  /* For this path res2 must be exactly 0 or 1 */
-    if( (pOp->p5 & SQLITE_KEEPNULL)!=0 ){
+    if( (pOp->p5 & SQLITE_KEEPNULL)==SQLITE_KEEPNULL ){
```

Comment:
Sorry, can you give an example?
In general:
0110 & 0100   -> 0100     no need to compare with 0100, can just do if(x!=0).
and
0010 & 0100   -> 0000    
I think i am misunderstanding something.